db/migrate/20170501201633_add_canceled_to_order_detail.rb

```diff
@@ -4,10 +4,11 @@ def up
     add_column :order_details, :canceled_by, :integer
     add_column :order_details, :canceled_reason, :string
 
-    Reservation.where.not(order_detail_id: nil).find_each do |reservation|
-      reservation.order_detail.update_attributes(canceled_at: reservation.canceled_at,
+    Reservation.where.not(order_detail_id: nil, canceled_at: nil).find_each do |reservation|
+      reservation.order_detail.assign_attributes(canceled_at: reservation.canceled_at,
```

Comment:
The save needs to `order_detail`.

I'm looking to see if this has different behavior than `update_columns`.